elucidator-db/src/backends/rtree.rs

```diff
@@ -95,11 +96,41 @@ impl Database for RTreeDatabase {
             designations: HashMap::new(),
         })
     }
-    fn from_path(_: &str) -> Result<Self> {
-        unimplemented!();
+    fn from_path(filename: &str) -> Result<Self> {
+        let sqlite = SqlDatabase::from_path(filename)?;
+        let designations = sqlite.get_designations();
+        let mds = sqlite.get_all_metadata()?;
+        let rtree = RTree::bulk_load(mds);
+        Ok(RTreeDatabase {
+            rtree,
+            designations,
+        })
     }
-    fn save_as(&self, _filename: &str) -> Result<()> {
-        unimplemented!();
+    fn save_as(&self, filename: &str) -> Result<()> {
+        let mut sqlite = SqlDatabase::new(Some(filename), None)?;
+
+        for (designation, designation_spec) in self.designations.iter() {
+            sqlite.insert_spec_text(&designation, &designation_spec.to_string())?;
+        }
+        let md_clones: Vec<&MetadataClone> = self.rtree.iter().collect();
+        let mds: Vec<Metadata> = md_clones.iter()
+            .map(|m| 
+                Metadata {
+                    xmin: m.xmin,
+                    xmax: m.xmax,
+                    ymin: m.ymin,
+                    ymax: m.ymax,
+                    zmin: m.zmin,
+                    zmax: m.zmax,
+                    tmin: m.tmin,
+                    tmax: m.tmax,
+                    designation: &m.designation,
+                    buffer: &m.buffer,
+                }
+            )
+            .collect(); 
+        sqlite.insert_n_metadata(&mds)?;
```

Comment:
Let's see if we can avoid copying the entire contents of the metadata and iteratively insert individual metadata items.